src/loaders/css-module-dts-loader/loader.ts

```diff
@@ -0,0 +1,125 @@
+import { createSourceFile, forEachChild, Node, ScriptTarget, SyntaxKind } from 'typescript';
+import { statSync } from 'fs';
+import { resolve, dirname } from 'path';
+const DtsCreator = require('typed-css-modules');
+const { getOptions } = require('loader-utils');
+const instances = require('ts-loader/dist/instances');
+
+type TSLoaderInstances = {
+	files: {
+		[key: string]: boolean;
+	}
+}
+
+type DtsResult = {
+	writeFile(): Promise<void>;
+}
+
+type DtsCreatorInstance = {
+	create(filePath: string, initialContents: boolean, clearCache: boolean): Promise<DtsResult>;
+}
+
+type Webpack = {
+	resourcePath: string;
+	async(): (error: Error | null, result: string, sourceMap?: string) => void;
+}
+
+const creator: DtsCreatorInstance = new DtsCreator();
+
+const mTimeMap = new Map<string, Date>();
+
+function generateDTSFile(filePath: string): Promise<void> {
```

Comment:
If you always need to return a Promise, you can start a chain with a `Promise.resolve`, it will neaten the following up to:
```ts
function generateDTSFile(filePath: string): Promise<void> {
  return Promise.resolve().then(() => {
    const { mtime } = statSync(filePath);
    const lastMTime = mTimeMap.get(filePath);

    if (!lastMTime || mtime > lastMTime) {
      mTimeMap.set(filePath, mtime);
      return creator.create(filePath, false, true)
        .then((content) => content.writeFile());
    }
});
```